### PR TITLE
Increase test coverage for (and update) `root_signing.py`

### PR DESCRIPTION
**Pull request by beeankha**
### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: tests/test_root.py [hunk 1]
```diff
@@ -259,3 +219,50 @@ def test_verify_existing_root_md():
...
+        SAMPLE_FINGERPRINT, SAMPLE_KEYVAL
+    )
+
+    assert gpg_verdict
+
+
+def test_sign_root_metadata_via_gpg():
+    if not SSLIB_AVAILABLE:
+        pytest.skip(
+            "--TEST SKIPPED⚠️ : Unable to use GPG key retrieval or "
```
Comment (dholth) [inline tests/test_root.py:239]: Nice emoji

**Comment by dholth** [comment]
Great cleanups. I tried installing gpg and couldn't get certain existing tests to pass; possibly would have to create a key locally first?

**Review by dholth — APPROVED** [review]
Thanks!

File: tests/test_root.py [hunk 2]
```diff
@@ -131,71 +132,31 @@ def test_gpg_signing_with_unknown_fingerprint():
...
+            "signing without securesystemslib and GPG."
+        )
+        return
 
     # Build a basic root metadata file with empty key_mgr delegation and one
     # root key, threshold 1, version 1.
     rmd = metadata_construction.build_root_metadata(
         root_version=1,
-        root_pubkeys=[SAMPLE_KEYVAL],
+        root_pubkeys=[SAMPLE_FINGERPRINT], # used to be SAMPLE_KEYVAL
```
Comment (awwad) [inline tests/test_root.py:147]: I think this is the key value, not a fingerprint.

File: tests/test_root.py [hunk 3]
```diff
@@ -148,54 +151,12 @@ def test_root_gen_sign_verify():
...
-        gpg_sig, SAMPLE_KEYVAL, canonical_signed_portion
-    )
-
-    print(
-        "--TEST SUCCESS✅: GPG signing (using GPG and securesystemslib) and "
-        "GPG signature verification (using only cryptography)"
-    )
+    # Verify the signature.
+    root_signing.sign_via_gpg(rmd)
+    securesystemslib.gpg.functions.verify_signature(rmd)
```
Comment (awwad) [inline tests/test_root.py:159]: I think we want to use authentication.verify_gpg_signature, not securesystemslib.gpg.functions.verify_signature.

File: conda_content_trust/root_signing.py [hunk 4]
```diff
@@ -220,33 +203,26 @@ def sign_via_gpg(data_to_sign, gpg_key_fingerprint, include_fingerprint=False):
...
-        )
+    _check_sslib_available()
 
     # Make sure it's the right format.
     if not is_a_signable(root_signable):
-        raise TypeError(
-            "Expected a signable dictionary; the given file "
-            + str(root_md_fname)
-            + " failed the check."
+        raise Exception(
```
Comment (dholth) [inline conda_content_trust/root_signing.py:213]: Should not raise plain non-specific Exception()

File: tests/test_root.py [hunk 5]
```diff
@@ -148,54 +151,12 @@ def test_root_gen_sign_verify():
     )
     rmd = signing.wrap_as_signable(rmd)
 
-    signed_portion = rmd["signed"]
+    # Sign it with the GPG key.
+    root_signing.sign_root_metadata_dict_via_gpg(rmd, SAMPLE_FINGERPRINT)
```
Comment (awwad) [inline tests/test_root.py:155]: Looks like a typo?  This is importing `sign_root_metadata_via_gpg` and then calling `sign_root_metadata_dict_via_gpg`.  Separately, it's also not using the reference imported in line 18, but instead just calling it from the `root_signing` module.

File: tests/test_root.py (hunk 3 above)
Comment (beeankha) [reply tests/test_root.py]: I've reverted the test to (mostly) how it was

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -58,7 +58,9 @@ jobs:
       - name: Setup environment
         run: |
           conda install -q -y -c defaults \
-            --file ./tests/requirements.txt
+            --file ./tests/requirements.txt \
+            conda-forge::securesystemslib
```
Comment (jezdez) [inline .github/workflows/tests.yml:62]: Let's add that to the test requirements

File: tests/requirements.txt [hunk 7]
```diff
@@ -3,4 +3,6 @@
 cryptography>=41.0.0
 pytest
 pytest-cov
+conda>=23.7
```
Comment (jezdez) [inline tests/requirements.txt:6]: That's a big jump, any specific reason?

Comment (dholth) [reply tests/requirements.txt:6]: It was installing conda 4

Comment (kenodegard) [reply tests/requirements.txt:6]: I tested this yesterday (see #77) and conda 22.11 is sufficient

Comment (kenodegard) [reply tests/requirements.txt]: continue here https://github.com/conda/conda-content-trust/pull/75#discussion_r1302039760

File: tests/requirements.txt [hunk 8]
```diff
@@ -3,4 +3,7 @@
 cryptography>=41.0.0
 pytest
 pytest-cov
-# securesystemslib ==0.13.1  # not available on defaults or conda-forge
+pytest-mock
+conda>23.7
```
[the remark this answers is not in the document]
Comment (dholth) [reply tests/requirements.txt:7]: Why do we care if we can run this code against an old version of conda?

**Review by dholth — APPROVED** [review]
LGTM

**Pull request merged**